File: haskell/raindrops/src/Raindrops.hs
module Raindrops (convert) where

convert :: Int -> String
convert n =
  case sounds of
    "" -> show n
    _ -> sounds
  where sound m s
          | n `mod` m == 0 = s
          | otherwise = ""
        sounds = sound 3 "Pling" <> sound 5 "Plang" <> sound 7 "Plong"

File: clojure/raindrops/src/raindrops.clj
(ns raindrops
  (:use [clojure.core]))

(defn convert [number]
  (let [sound (with-out-str
                (when (zero? (mod number 3))
                  (print "Pling"))
                (when (zero? (mod number 5))
                  (print "Plang"))
                (when (zero? (mod number 7))
                  (print "Plong")))]
    (if (empty? sound)
      (str number)
      sound)))
